Framework/Core/include/Framework/DataAllocator.h

```diff
@@ -56,7 +56,7 @@ class DataAllocator
                 const AllowedOutputRoutes &routes);
 
   DataChunk newChunk(const Output&, size_t);
-  DataChunk newChunk(OutputRef const& ref, size_t size) { return newChunk(getOutputByBind(ref), size); }
+  DataChunk newChunk(OutputRef&& ref, size_t size) { return newChunk(getOutputByBind(std::move(ref)), size); }
```

Comment:
@ktf, the `Stack` member of `OutputRef` needs to be transferred over to the `Output` object. Since there is neither copy constructor nor assignment operator I have to use move assignment. That's the reason for the change. I put a note in front of each of the methods that the `OutputRef` argument is expected to be rvalue.